ginga/doc/download_doc.py

```diff
@@ -0,0 +1,179 @@
+    Download is skipped if content already exists.
+
+    Parameters
+    ----------
+    rtd_version : str or `None`
+        RTD version to download; e.g., "latest", "stable", or "v2.6.0".
+        If not given, download closest match to software version.
+
+    kwargs : dict
+        Keywords for ``urlretrieve()``.
+
+    Returns
+    -------
+    index_html : str
+        Path to local "index.html".
+
+    """
+    if rtd_version is None:
+        rtd_version = _find_rtd_version()
+
+    data_path = os.path.dirname(
+        _find_pkg_data_path('help.html', package='ginga.doc'))
+    index_html = os.path.join(data_path, 'index.html')
+
+    # There is a previous download of documentation; Do nothing.
+    # There is no check if downloaded version is outdated; The idea is that
+    # this folder would be empty again when installing new version.
+    if os.path.isfile(index_html):
+        return index_html
+
+    url = ('https://readthedocs.org/projects/ginga/downloads/htmlzip/'
+           '{}/'.format(rtd_version))
+    local_path = urllib.request.urlretrieve(url, **kwargs)[0]
+
+    with zipfile.ZipFile(local_path, 'r') as zf:
+        zf.extractall(data_path)
+
+    # RTD makes an undesirable sub-directory, so move everything there
+    # up one level and delete it.
+    subdir = os.path.join(data_path, 'ginga-{}'.format(rtd_version))
+    for s in os.listdir(subdir):
+        src = os.path.join(subdir, s)
+        if os.path.isfile(src):
+            shutil.copy(src, data_path)
+        else:  # directory
+            shutil.copytree(src, os.path.join(data_path, s))
+    shutil.rmtree(subdir)
+
+    if not os.path.isfile(index_html):
+        raise OSError(
+            '{} is missing; Ginga doc download failed'.format(index_html))
+
+    return index_html
+
+
+def get_doc(logger=None, plugin=None, reporthook=None):
+    """
+    Return URL to documentation. Attempt download if does not exist.
+
+    Parameters
+    ----------
+    logger : obj or `None`
+        Ginga logger.
+
+    plugin : obj or `None`
+        Plugin object. If given, URL points to plugin doc directly.
+        If this function is called from within plugin class,
+        pass ``self`` here.
+
+    reporthook : callable or `None`
+        Report hook for ``urlretrieve()``.
+
+    Returns
+    -------
+    url : str or `None`
+        URL to local documentation, if available.
+
+    """
+    from ginga.GingaPlugin import GlobalPlugin, LocalPlugin
+
+    if isinstance(plugin, GlobalPlugin):
+        # plugin_page = 'plugins_global'
+        plugin_name = str(plugin)
+    elif isinstance(plugin, LocalPlugin):
+        # plugin_page = 'plugins_local'
+        plugin_name = str(plugin)
+    else:
+        # plugin_page = None
+        plugin_name = None
+
+    try:
+        index_html = _download_rtd_zip(reporthook=reporthook)
+
+    # Download failed, use online resource
+    except Exception as e:
+        url = 'https://ginga.readthedocs.io/en/latest/'
```

Comment:
Can this URL be set to go to the plugin section that the user is asking help for?